src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -536,6 +560,14 @@ def validate(self, data):
                 f"Number of widgets must be less than {Dashboard.MAX_WIDGETS}"
             )
 
+        permissions = data.get("permissions")
+        if permissions and self.instance:
+            currentUser = self.context["request"].user
+            if self.instance.created_by_id != currentUser.id:
+                raise serializers.ValidationError(
+                    "Only the Dashboard Creator may modify Dashboard Edit Access"
+                )
```

Comment:
I just realized, this will trigger if a user updates the dashboard (e.g. they change the title) and the dashboard that we're sending from the frontend has "permissions" in its state, right?

1. We should feature flag this so it doesn't affect customers inadvertently
2. Either we should check that the request _is_ actually changing the permissions, or we should ensure that the frontend _only_ sends this when they change the permissions selector

Reply:
In the case that the user is just updating the dashboard's title, `permissions` will be `None`, so it wouldn't go through the `if` condition. Do you mean that there needs to be a check before the `data.get("permissions")` statement?